staging/src/k8s.io/legacy-cloud-providers/gce/gce_disks.go

```diff
@@ -620,6 +624,50 @@ func (g *Cloud) DisksAreAttached(diskNames []string, nodeName types.NodeName) (m
 	return attached, nil
 }
 
+// BulkDisksAreAttached is a batch function to check if all corresponding disks are attached to the
+// nodes specified with nodeName.
+func (g *Cloud) BulkDisksAreAttached(diskByNodes map[types.NodeName][]string) (map[types.NodeName]map[string]bool, error) {
+	instanceNames := []string{}
+	for nodeName := range diskByNodes {
+		instanceNames = append(instanceNames, mapNodeNameToInstanceName(nodeName))
+	}
+
+	// List all instances with the given instance names
+	// Then for each instance listed, add the disks attached to that instance to a map
+	listedInstances, err := g.getInstancesByNames(instanceNames)
+	listedInstanceNamesToDisks := make(map[string][]*compute.AttachedDisk)
+	for _, instance := range listedInstances {
+		listedInstanceNamesToDisks[instance.Name] = instance.Disks
+	}
+	if err != nil {
+		if err == cloudprovider.InstanceNotFound {
+			notFound := []string{}
+			for _, instanceName := range instanceNames {
```

Comment:
not sure what this inner error loop is doing, `getInstanceByNames` always returns nil for the list of instances if it returns an error

Reply:
I changed that function so that if the error is cloudprovider.InstanceNotFound, it doesn't return nil but rather the list of instances it could find.